senders/msteams/msteams.go

```diff
@@ -0,0 +1,207 @@
+package msteams
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/moira-alert/moira"
+	"github.com/russross/blackfriday/v2"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+)
+
+// Sender implements moira sender interface via MS Teams
+type Sender struct {
+	frontURI string
+	logger   moira.Logger
+	location *time.Location
+	client   *http.Client
+}
+
+// Init initialises settings required for full functionality
+func (sender *Sender) Init(senderSettings map[string]string, logger moira.Logger, location *time.Location, dateTimeFormat string) error {
+	sender.logger = logger
+	sender.location = location
+	sender.frontURI = senderSettings["front_uri"]
+	sender.client = &http.Client{
+		Timeout: time.Duration(30) * time.Second,
+	}
+	return nil
+}
+
+// SendEvents implements Sender interface Send
+func (sender *Sender) SendEvents(events moira.NotificationEvents, contact moira.ContactData, trigger moira.TriggerData, plot []byte, throttled bool) error {
+
+	err := sender.isValidWebhookURL(contact.Value)
+	if err != nil {
+		return fmt.Errorf("invalid msteams webhook url: %s", err.Error())
+	}
+
+	request, err := sender.buildRequest(events, contact, trigger, plot, throttled)
+
+	if request != nil {
+		defer request.Body.Close()
+	}
```

Comment:
As per documentation, `The request Body, if non-nil, will be closed by the underlying Transport, even on errors.` So we don't need this.

I understand that you copied this code from our own webhooks, but we can make it better (and also raise an issue to make webhooks code better).